app/objects/c_ability.py

```diff
@@ -1,15 +1,51 @@
 import os
 from base64 import b64decode
 
+import marshmallow as ma
+
 from app.objects.interfaces.i_object import FirstClassObjectInterface
-from app.objects.secondclass.c_parser import Parser
-from app.objects.secondclass.c_requirement import Requirement
-from app.objects.secondclass.c_variation import Variation
+from app.objects.secondclass.c_parser import Parser, ParserSchema
+from app.objects.secondclass.c_requirement import Requirement, RequirementSchema
+from app.objects.secondclass.c_variation import Variation, VariationSchema
 from app.utility.base_object import BaseObject
+from app.utility.base_world import AccessSchema
+
+
+class AbilitySchema(ma.Schema):
+    ability_id = ma.fields.String()
+    tactic = ma.fields.String()
+    technique_name = ma.fields.String()
+    technique_id = ma.fields.String()
+    name = ma.fields.String()
+    description = ma.fields.String()
+    cleanup = ma.fields.List(ma.fields.String())
+    executor = ma.fields.String()
+    platform = ma.fields.String()
+    payloads = ma.fields.List(ma.fields.String())
+    parsers = ma.fields.List(ma.fields.Nested(ParserSchema))
+    requirements = ma.fields.List(ma.fields.Nested(RequirementSchema))
+    privilege = ma.fields.String()
+    timeout = ma.fields.Int()
+    repeatable = ma.fields.Bool()
+    language = ma.fields.String()
+    code = ma.fields.String()
+    build_target = ma.fields.String()
+    variations = ma.fields.List(ma.fields.Nested(VariationSchema))
+    buckets = ma.fields.List(ma.fields.String())
+    additional_info = ma.fields.Dict(keys=ma.fields.String(), values=ma.fields.String())
+    access = ma.fields.Nested(AccessSchema)
+    test = ma.fields.String()
+
+    @ma.post_load
+    def build_ability(self, data, **_):
+        return Ability(**data)
 
 
 class Ability(FirstClassObjectInterface, BaseObject):
 
+    schema = AbilitySchema()
+    display_schema = AbilitySchema(exclude=['repeatable', 'language', 'code', 'build_target'])  # may need to fix for id=self.unique
```

Comment:
before approving need to address my comment here, as in the current display scheme we are displaying `self.unique` as both `unique` and as `id`.  Do we still need `id`, or can it be dropped?